blocks/v2-icon-cards/v2-icon-cards.js

```diff
@@ -1,26 +1,145 @@
-import { createElement, variantsClassesToBEM } from '../../scripts/common.js';
+import { createElement, decorateIcons, getTextLabel, variantsClassesToBEM, isMobileViewport } from '../../scripts/common.js';
 
 const blockName = 'v2-icon-cards';
-const variantClasses = ['no-background', 'alt-font-size'];
+const variantClasses = ['no-background', 'alt-font-size', 'slider'];
 
-export default async function decorate(block) {
-  variantsClassesToBEM(block.classList, variantClasses, blockName);
-  const rows = [...block.querySelectorAll(':scope > div')];
+function numberCardsToNavigate(block) {
+  const isMobile = isMobileViewport();
+  if (isMobile) {
+    return 1;
+  }
+
+  if (block.classList.contains('slider-slide-2')) {
+    return 2;
+  }
+  if (block.classList.contains('slider-slide-3')) {
+    return 3;
+  }
+  if (block.classList.contains('slider-slide-4')) {
+    return 4;
+  }
+
+  return 1;
+}
+
+function mergeColumnsInOneRow(block) {
   const columns = [...block.querySelectorAll(':scope > div > div')];
 
+  if (columns.length > 5) {
+    const row = document.createElement('div');
+
+    row.append(...columns);
+    block.innerHTML = '';
+    block.append(row);
+  }
+}
+
+function setSliderNavButtonEvents(block) {
+  const numberCardsToNavigatePerClick = numberCardsToNavigate(block);
+  const sliderNav = block.querySelector('.v2-icon-cards__slider-nav');
```

Comment:
It's ok as it is, but the selector should go with `${blockname}__[something]` isn't it?